services/skipchain/skipchain.go

```diff
@@ -0,0 +1,321 @@
+package skipchain
+
+import (
+	"crypto/rand"
+	"errors"
+
+	"fmt"
+
+	"bytes"
+
+	"sync"
+
+	"github.com/dedis/cothority/lib/cosi"
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+// ServiceName can be used to refer to the name of this service
+const ServiceName = "Skipchain"
+
+func init() {
+	sda.RegisterNewService(ServiceName, newSkipchainService)
+	skipchainSID = sda.ServiceFactory.ServiceID(ServiceName)
+}
+
+var skipchainSID sda.ServiceID
+
+// Service handles adding new SkipBlocks
+type Service struct {
+	*sda.ServiceProcessor
+	// SkipBlocks points from SkipBlockID to SkipBlock but SkipBlockID is not a valid
+	// key-type for maps, so we need to cast it to string
+	SkipBlocks map[string]*SkipBlock
```

Comment:
Right. But if we have more than one encryption suite, we'll still have the same problem. So it turns out that it may be not the best way to **use** the suite hashes for this, **because** it's not a fixed size. We may be better off saying that SkipBlockID are sha256 IDs and that's it, unrelated to whatever suite we use. I think it's quite common to fix a hashing algorithm for stuff like that. We won't get bitten in the foot when we will want to change to another suite. Like in `lib/sda`, we sticked to `UUID` which are `[16]byte`. What do you think ?